providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/resolver/process/storage/connector/grpc/GrpcStreamConnector.java

```diff
@@ -150,11 +174,18 @@ static void observeEventStream(final BlockingQueue<StreamPayload> writeTo,
                 final SyncFlagsResponse flagsResponse = response.getSyncFlagsResponse();
                 String data = flagsResponse.getFlagConfiguration();
                 log.debug("Got stream response: " + data);
+
                 if (!writeTo.offer(
-                        new StreamPayload(StreamPayloadType.DATA, data))) {
+                        new QueuePayload(QueuePayloadType.DATA, data, metadata))) {
                     log.error("Stream writing failed");
                 }
 
+                if (metadataException != null) {
+                    // if we somehow are connected but the metadata call failed, something strange
+                    // happened
+                    log.error("Stream connected but getMetadata RPC failed", metadataException);
```

Comment:
Should we reset `metadataException` to not log in every iteration?

Reply:
Oh actually we already so this, since this exception is declared inside the while loop.